src/components/PopupWithForm.js

```diff
@@ -6,29 +6,36 @@ export default class PopupWithForm extends Popup {
     this._popupForm = this._popup.querySelector(".popup__form");
     this._inputs = this._popupForm.querySelectorAll(".popup__input");
     this._handleFormSubmit = handleFormSubmit;
+    this._buttonText = this._popupForm.querySelector(".popup__button");
     this.setEventListeners();
   }
 
+  saving(show) {
+    console.log(this._buttonText);
+    show
+      ? (this._buttonText.textContent = "Saving...")
+      : (this._buttonText.textContent = "Save");
+  }
+
   setEventListeners = () => {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
       this._handleFormSubmit(this._getInputValues());
-      this.close();
       evt.target.reset();
```

Comment:
Please, delete `evt.target.reset();` from `PopupConfirm` and `PopupWithForm`

You need to reset the form only in block `then` after a successful response from the server because if there is an error the user should be able to send the request again without having to type all the info again.